test/parallel/test-stringbytes-external.js

```diff
@@ -119,3 +119,68 @@ var PRE_3OF4_APEX = Math.ceil((EXTERN_APEX / 4) * 3) - RADIOS;
   assert.equal(a, b);
   assert.equal(b, c);
 })();
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString();
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('ascii');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('utf8');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength * 2 + 2).fill().toString('utf16le');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('binary');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('base64');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('hex');
+  }, Error);
+
+  try {
+    new Buffer(kStringMaxLength * 4);
+  } catch(e) {
+    assert.equal(e.message, 'Buffer allocation failed - process out of memory');
+    console.log('skip non-throwing toString tests due to memory confinements');
```

Comment:
@thefourtheye This check is only done because you can't distinguish between toString error causes in the following tests which would fail in a memory confined environment

Changed log format to TAP skip.